numpy/core/src/multiarray/ctors.c

```diff
@@ -2005,9 +2011,31 @@ PyArray_FromArray(PyArrayObject *arr, PyArray_Descr *newtype, int flags)
             return NULL;
         }
 
-        if (flags & NPY_ARRAY_UPDATEIFCOPY)  {
+        if (flags & NPY_ARRAY_UPDATEIFCOPY) {
+            /* This is the ONLY place the NPY-ARRAY_UPDATEIFCOPY flag
+             * is checked to call PyArray_SetUpdateIfCopyBase
+             * Deprecate this (and remove the `else` from the following
+             * condition) once the flag itself is deprecated
+             */
+
+            if (DEPRECATE("NPY_ARRAY_UPDATEIFCOPY (possibly via "
+                "NPY_ARRAY_INOUT_ARRAY or NPY_ARRAY_INOUT_FARRAY) is "
```

Comment:
Probably simpler to phrase this as "NPY_ARRAY_UPDATEIFCOPY, NPY_ARRAY_INOUT_ARRAY, and NPY_ARRAY_INOUT_FARRAY are deprecated, use ..."